pkg/getter/ocigetter.go

```diff
@@ -39,22 +39,30 @@ func (g *OCIGetter) Get(href string, options ...Option) (*bytes.Buffer, error) {
 func (g *OCIGetter) get(href string) (*bytes.Buffer, error) {
 	client := g.opts.registryClient
 
-	ref := strings.TrimPrefix(href, "oci://")
+	ref := strings.TrimPrefix(href, fmt.Sprintf("%s://", registry.OCIScheme))
+
+	var pullOpts []registry.PullOption
+	requestingProv := strings.HasSuffix(ref, ".prov")
+	if requestingProv {
+		ref = strings.TrimSuffix(ref, ".prov")
+		pullOpts = append(pullOpts,
+			registry.PullOptWithChart(false),
```

Comment:
What happens if PullOptWithChart is true or omitted?

Reply:
If omitted, defaults to true. If true, then true. I had considered having an option such as `PullOptWithoutChart` for the prov-only case, but it seemed strange if one option was "With" and the other "Without". Happy to go that route though

Reply:
To clarify, @adamreese and I were trying to determine what would happen if someone were to provide both `registry.PullOptWithChart(true)` and `registry.PullOptWithProv(true)` in the same request. As we were reading through the test cases, we did not see any scenario testing for that case.

We couldn't tell if the `result` on line 57 would contain both the chart and the provenance file. [The PullResult](https://github.com/helm/helm/blob/593eac8876cb16e95a2fd8bd6886dbdb3ee36a68/internal/experimental/registry/client.go#L200-L206) does appear to have both fields, and looking at the code more closely [it does appear both fields are populated](https://github.com/helm/helm/blob/593eac8876cb16e95a2fd8bd6886dbdb3ee36a68/internal/experimental/registry/client.go#L334-L359), assuming both options are set.